src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java

```diff
@@ -581,6 +583,35 @@ public boolean isEnabled() {
         }
     };
 
+    @SuppressWarnings("serial") // anonymous class
+    private final Action openPopupOrhighlightLast = new AbstractAction() {
+        @Override
+        public void actionPerformed(final ActionEvent e) {
+            final JComboBox<?> comboBox = (JComboBox<?>) e.getSource();
+            if (!comboBox.isPopupVisible()) {
+                comboBox.setPopupVisible(true);
+            } else {
+                final int size = listBox.getModel().getSize();
+                listBox.setSelectedIndex(size - 1);
+                listBox.ensureIndexIsVisible(size - 1);
+            }
+        }
+    };
+
+    @SuppressWarnings("serial") // anonymous class
+    private final Action openPopupOrhighlightFirst = new AbstractAction() {
```

Comment:
may be openPopupOrHighlightFirst (uppercase H)

Reply:
ok